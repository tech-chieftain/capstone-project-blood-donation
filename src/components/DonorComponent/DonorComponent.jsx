import React from 'react';
import { PropTypes } from 'prop-types';
import name from '../images/donorcomponent/name.svg';
import blood from '../images/donorcomponent/blood_type.svg';
import phone from '../images/donorcomponent/phone.svg';
import addresIco from '../images/donorcomponent/location.svg';
import emailIco from '../images/donorcomponent/email.svg';
import DonorCard from '../DonorCard/DonorCard';

function DonorComponent({ donorName, bloodType, phoneNumber, address, email }) {
  return (
    <div>
      <DonorCard title={donorName} cardIcon={name} />
      <DonorCard title={bloodType} cardIcon={blood} />
      <DonorCard title={phoneNumber} cardIcon={phone} />
      <DonorCard title={address} cardIcon={addresIco} />
      <DonorCard title={email} cardIcon={emailIco} />
    </div>
  );
}

DonorComponent.propTypes = {
  donorName: PropTypes.element.isRequired,
  bloodType: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  email: PropTypes.element.isRequired,
};
export default DonorComponent;
