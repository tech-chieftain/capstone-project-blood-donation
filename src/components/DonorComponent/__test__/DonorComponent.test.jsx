import React from 'react';
import renderer from 'react-test-renderer';
import DonorComponent from '../DonorComponent';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DonorComponent
        donorName="Emad Alhattami"
        bloodType="A-"
        phoneNumber="774373827"
        address="Sana'a"
        email="emad@gmail.com"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
