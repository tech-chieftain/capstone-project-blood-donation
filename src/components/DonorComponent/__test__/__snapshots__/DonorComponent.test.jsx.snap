// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="px-1  bg-gray-200 sm:w-full"
  >
    <div
      className="flex"
    >
      <img
        alt="icon"
        className="w-6 h-6 px-1 my-3 mr-2"
        src="name.svg"
      />
      <span
        className="pt-3 text-lg sm:text-2xl md:text-2xl font-bold"
      >
         
        Emad Alhattami
         
      </span>
    </div>
  </div>
  <div
    className="px-1  bg-gray-200 sm:w-full"
  >
    <div
      className="flex"
    >
      <img
        alt="icon"
        className="w-6 h-6 px-1 my-3 mr-2"
        src="blood_type.svg"
      />
      <span
        className="pt-3 text-lg sm:text-2xl md:text-2xl font-bold"
      >
         
        A-
         
      </span>
    </div>
  </div>
  <div
    className="px-1  bg-gray-200 sm:w-full"
  >
    <div
      className="flex"
    >
      <img
        alt="icon"
        className="w-6 h-6 px-1 my-3 mr-2"
        src="phone.svg"
      />
      <span
        className="pt-3 text-lg sm:text-2xl md:text-2xl font-bold"
      >
         
        774373827
         
      </span>
    </div>
  </div>
  <div
    className="px-1  bg-gray-200 sm:w-full"
  >
    <div
      className="flex"
    >
      <img
        alt="icon"
        className="w-6 h-6 px-1 my-3 mr-2"
        src="location.svg"
      />
      <span
        className="pt-3 text-lg sm:text-2xl md:text-2xl font-bold"
      >
         
        Sana'a
         
      </span>
    </div>
  </div>
  <div
    className="px-1  bg-gray-200 sm:w-full"
  >
    <div
      className="flex"
    >
      <img
        alt="icon"
        className="w-6 h-6 px-1 my-3 mr-2"
        src="email.svg"
      />
      <span
        className="pt-3 text-lg sm:text-2xl md:text-2xl font-bold"
      >
         
        emad@gmail.com
         
      </span>
    </div>
  </div>
</div>
`;
